browser/search_engines/search_engine_tracker.cc

```diff
@@ -176,6 +185,10 @@ void SearchEngineTracker::RecordSwitchP3A(const GURL& url) {
     answer = SearchEngineSwitchP3AMapAnswer(url, previous_search_url_);
     previous_search_url_ = url;
     switch_record_.Add(static_cast<int>(answer));
+
+    if (url.DomainIs(kBraveDomain)) {
+      brave_search_conversion::p3a::RecordDefaultEngineChange(local_state_);
```

Comment:
why not use profile_prefs?

Reply:
Please see https://github.com/brave/brave-core/pull/13566#discussion_r888605271